### ISSUE TITLE
False positive for `explicit_iter_loop` with large arrays

### ISSUE THREAD

**Issue by apoelstra** [Manishearth/rust-clippy#419]
On the code
```rust
fn main() {
    let mut arr = [0u8; 64];
    for elem in arr.iter_mut() {
        *elem = 1;
    }
}
```
clippy warns
```
src/main.rs:4:5: 6:6 warning: it is more idiomatic to loop over `&mut arr` instead of `arr.iter_mut()`, #[warn(explicit_iter_loop)] on by default
src/main.rs:4     for elem in arr.iter_mut() {
src/main.rs:5         *elem = 1;
src/main.rs:6     }
src/main.rs:4:5: 6:6 help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#explicit_iter_loop
```
But if you change the code to use `&mut arr` in place of `arr.iter_mut()` you get a compiler error:
```
src/main.rs:4:5: 6:6 error: the trait `core::iter::Iterator` is not implemented for the type `[u8; 64]` [E0277]
src/main.rs:4     for elem in &mut arr {
src/main.rs:5         *elem = 1;
src/main.rs:6     }
src/main.rs:4:5: 6:6 help: run `rustc --explain E0277` to see a detailed explanation
src/main.rs:4:5: 6:6 note: `[u8; 64]` is not an iterator; maybe try calling `.iter()` or a similar method
error: aborting due to previous error
```
Note that this does work for small arrays, i.e. the ones that have `Iterator` implemented on them by the stdlib.

**Comment by llogiq** [comment]
That seems to be a regression – we used to have a check for array sizes somewhere IIRC.

**Comment by mcarton** [comment]
[We do](https://github.com/Manishearth/rust-clippy/blob/2abb775de5bed06722ef0d9d14488d6e8777842f/tests/compile-fail/for_loop.rs#L215), so I’m closing. Feel free to reopen if you still have a problem.
